### PR TITLE
doc: prerequisites fix for cephFS mount

### PR DESCRIPTION
**Pull request by nmshelke**
Fixes: https://tracker.ceph.com/issues/53054

Signed-off-by: Nikhilkumar Shelke <nshelke@redhat.com>

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests
- Teuthology
  - [ ] Completed teuthology run
  - [x] No teuthology test necessary (e.g., documentation)

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: doc/cephfs/mount-prerequisites.rst [hunk 1]
```diff
@@ -6,6 +6,8 @@ You can use CephFS by mounting it to your local filesystem or by using
 FUSE`_. Both have their own advantages. Read the following section to
 understand more about both of these ways to mount CephFS.
 
+Mounting CephFS requires superuser privileges.
+
```
Comment (rishabh-d-dave) [inline doc/cephfs/mount-prerequisites.rst:10]: The change feels out of place. I think following would be better.

```suggestion
FUSE`_. Both have their own advantages and both require superuser
privileges to mount. Read the following section to understand more about
both of these ways to mount CephFS.
```

Also, replacing [`fuse.conf option`section](https://docs.ceph.com/en/pacific/cephfs/mount-using-fuse/#fuse-conf-option) by a note point (for informing and convenience of `ceph-fuse` users) would be great.

**Comment by nmshelke** [comment]
jenkins retest this please

**Comment by nmshelke** [comment]
jenkins retest this please

File: doc/cephfs/mount-prerequisites.rst (hunk 1 above)
Comment (nmshelke) [reply doc/cephfs/mount-prerequisites.rst]: Done.

**Comment by kotreshhr** [comment]
https://ceph--43886.org.readthedocs.build/en/43886/cephfs/mount-prerequisites/
https://ceph--43886.org.readthedocs.build/en/43886/cephfs/mount-using-fuse/

**Review by kotreshhr — APPROVED** [review]

File: doc/cephfs/mount-using-fuse.rst [hunk 2]
```diff
@@ -9,17 +9,10 @@ clients can be more manageable, especially while upgrading CephFS.
...
 Go through the prerequisites required by both, kernel as well as FUSE mounts,
 in `Mount CephFS: Prerequisites`_ page.
 
-``fuse.conf`` option
---------------------
-
-#. If you are mounting Ceph with FUSE not as superuser/root user/system admin
-   you would need to add the option ``user_allow_other`` to ``/etc/fuse.conf``
-   (under no section in the conf).
+.. note:: Mounting CephFS using FUSE requires superuser privileges.
```
Comment (rishabh-d-dave) [inline doc/cephfs/mount-using-fuse.rst:15]: nit: To emphasize that this change is relatively recent.
```suggestion
.. note:: Mounting CephFS using FUSE requires superuser privileges now.
```

**Review by rishabh-d-dave — APPROVED** [review]

**Review by joscollin — APPROVED** [review]
The `fuse.conf option` seems invalid, may be too old. Only root can use "--options" now. So it can be removed.

File: doc/cephfs/mount-using-fuse.rst (hunk 2 above)
Comment (batrick) [reply doc/cephfs/mount-using-fuse.rst:15]: Go further and explain why it requires superuser privileges.

**Review by batrick — CHANGES_REQUESTED** [review]

File: doc/cephfs/mount-using-fuse.rst (hunk 2 above)
Comment (nmshelke) [reply doc/cephfs/mount-using-fuse.rst]: Done

File: doc/cephfs/mount-using-fuse.rst [hunk 3]
```diff
@@ -9,17 +9,10 @@ clients can be more manageable, especially while upgrading CephFS.
...
 Go through the prerequisites required by both, kernel as well as FUSE mounts,
 in `Mount CephFS: Prerequisites`_ page.
 
-``fuse.conf`` option
---------------------
-
-#. If you are mounting Ceph with FUSE not as superuser/root user/system admin
-   you would need to add the option ``user_allow_other`` to ``/etc/fuse.conf``
-   (under no section in the conf).
+.. note:: Mounting CephFS using FUSE requires superuser privileges to trim dentries.
```
Comment (batrick) [inline doc/cephfs/mount-using-fuse.rst:15]:
```suggestion
.. note:: Mounting CephFS using FUSE requires superuser privileges to trim dentries by issuing a remount of itself.
```

**Review by batrick — CHANGES_REQUESTED** [review]

File: doc/cephfs/mount-using-fuse.rst (hunk 3 above)
Comment (nmshelke) [reply doc/cephfs/mount-using-fuse.rst]: Done

**Review by batrick — APPROVED** [review]

**Review by vshankar — APPROVED** [review]

**Pull request merged**